central/sensor/service/connection/ratelimit_manager.go

```diff
@@ -0,0 +1,248 @@
+package connection
+
+import (
+	"container/heap"
+	"math"
+	"time"
+
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/grpc/ratelimit"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	boostInitSyncRateLimit      = 50
+	defaultTopCandidateClusters = 3
+	defaultRatePeriod           = 10 * time.Minute
+	minimumRatePeriod           = 10 * time.Second
+)
+
+// Inspiration is taken from EARRRL:
+// https://blog.jnbrymn.com/2021/03/18/estimated-average-recent-request-rate-limiter.html
+//
+// A shorter period will cause older rates to diminish more rapidly, while
+// a longer period will result in older rates retaining greater importance.
+// The tick is set to 1s. The minimum period is 10s.
+type clusterMsgRate struct {
+	mutex sync.Mutex
+
+	// index is required for Heap.
+	index     int
+	clusterID string
+
+	lastUpdate  int64
+	halfPeriod  float64
+	ratePerSec  float64
+	accumulator float64
+}
+
+func (cmr *clusterMsgRate) recvMsg() {
+	cmr.mutex.Lock()
+	defer cmr.mutex.Unlock()
+
+	now := time.Now().Unix()
```

Comment:
I tried this. It is still better with `if`.

```
BenchmarkClusterRateReceiveMsgNano/time_ticks-12         	 7165490	       169.9 ns/op
BenchmarkClusterRateReceiveMsgNano/no_time_ticks-12      	 6870762	       167.6 ns/op
```
So around `~165-170 ns/op`.

With `if` it's `~50ns` faster. This is beneficial if we have many messages within 1 second.

So, I went with a solution without `if`, but it still uses `seconds`. The logic is the same, because `exp(0) = 1` and it will not change the accumulator value if `1s` didn't pass.